streams/src/test/java/org/apache/kafka/test/MockProcessorNode.java

```diff
@@ -60,7 +61,7 @@ public void init(final InternalProcessorContext<KOut, VOut> context) {
 
     @Override
     public void process(final Record<KIn, VIn> record) {
-        processor().process(record);
+        mockProcessor.process(record);
```

Comment:
Oh, now I see. Why not just capture the result of `ProcessorAdapter.adapt(mockProcessor)` in the constructor and use that for the `this.mockProcessor` field instead? That way it would already be the right type. That's basically what we were doing before, although I agree we don't need to have that method to expose the processor on the super class.

Reply:
Got it. Replaced the initialization with: `this.mockProcessor = (MockApiProcessor<...>) ProcessorAdapter.adapt(mockProcessor);`
Let me now if this look good now.

Reply:
Fallback to `MockProcessor` instead of `MockApiProcessor` as there is no need to migrate this I think.